internal/hcp/registry/hcl.go

```diff
@@ -210,18 +197,26 @@ func (h *HCLRegistry) registerAllComponents() hcl.Diagnostics {
 				continue
 			}
 
-			buildName := source.String()
-			if build.Name != "" {
-				buildName = fmt.Sprintf("%s.%s", build.Name, buildName)
+			if build.Name == "" {
+				diags = append(diags, &hcl.Diagnostic{
+					Severity: hcl.DiagError,
+					Summary:  "Build name conflicts",
```

Comment:
I would suggest pointing out at the lack of a build name here instead as the summary
```suggestion
					Summary:  "Missing mandatory build name",
```